iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/schema/ttl/TTLCache.java

```diff
@@ -0,0 +1,275 @@
+   */
+  public void unsetTTL(String[] nodes) {
+    if (nodes.length < 2) {
+      return;
+    } else if (nodes.length == 2) {
+      // if path equals to root.**, then unset it to configured ttl
+      if (nodes[0].equals(IoTDBConstant.PATH_ROOT)
+          && nodes[1].equals(IoTDBConstant.MULTI_LEVEL_PATH_WILDCARD)) {
+        ttlCacheTree.getChild(IoTDBConstant.MULTI_LEVEL_PATH_WILDCARD).ttl =
+            CommonDescriptor.getInstance().getConfig().getDefaultTTLInMs();
+        return;
+      }
+    }
+    CacheNode parent = ttlCacheTree;
+    int index = 0;
+    boolean usefulFlag;
+    CacheNode parentOfSubPathToBeRemoved = null;
+    for (int i = 1; i < nodes.length; i++) {
+      usefulFlag = !parent.getChildren().isEmpty() || parent.ttl != NULL_TTL;
+      CacheNode child = parent.getChild(nodes[i]);
+      if (child == null) {
+        // there is no matching path on ttl cache tree
+        return;
+      }
+      if (usefulFlag) {
+        parentOfSubPathToBeRemoved = parent;
+        index = i;
+      }
+      parent = child;
+    }
+    // currently, parent is the leaf node of the path to be removed
+    if (parent.ttl != NULL_TTL) {
+      ttlCount--;
+    }
+
+    if (!parent.getChildren().isEmpty()) {
+      // node to be removed is internal node, then just reset its ttl
+      parent.ttl = NULL_TTL;
+      return;
+    }
+
+    // node to be removed is leaf node, then remove corresponding node of this path from cache tree
+    if (parentOfSubPathToBeRemoved != null) {
+      parentOfSubPathToBeRemoved.removeChild(nodes[index]);
+    }
+  }
+
+  /**
+   * Get ttl from cache tree. Return the TTL of the node closest to the path leaf node that has a
+   * TTL which is not NULL_TTL.
+   *
+   * @param nodes should be prefix path or specific device path without wildcard
+   */
+  public long getTTL(String[] nodes) {
+    long ttl = ttlCacheTree.ttl;
+    CacheNode parent = ttlCacheTree;
+    for (int i = 1; i < nodes.length; i++) {
+      CacheNode child = parent.getChild(IoTDBConstant.MULTI_LEVEL_PATH_WILDCARD);
+      ttl = child != null ? child.ttl : ttl;
+      parent = parent.getChild(nodes[i]);
+      if (parent == null) {
+        break;
+      }
+    }
+    ttl = parent != null && parent.ttl != NULL_TTL ? parent.ttl : ttl;
+    return ttl;
+  }
+
+  public Map<String, Long> getAllTTLUnderOneNode(String[] nodes) {
+    Map<String, Long> pathTTLMap = new HashMap<>();
+    CacheNode node = ttlCacheTree;
+    for (int i = 1; i < nodes.length; i++) {
+      node = node.getChild(nodes[i]);
+      if (node == null) {
+        return pathTTLMap;
+      }
+    }
+
+    // get all ttl under current node
+    dfsCacheTree(pathTTLMap, new StringBuilder(new PartialPath(nodes).getFullPath()), node);
+    return pathTTLMap;
+  }
+
+  /**
+   * Return the ttl of path. If the path does not exist, it means that the TTL is not set, and
+   * return NULL_TTL.
+   */
+  public long getNodeTTL(String[] nodes) {
```

Comment:
I suggest changing `getTTL` and `getNodeTTL` to `getCloestTTL` and `getLastNodeTTL` to make them more distinguishable.

Reply:
Resolved.